src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -177,6 +223,45 @@ public class PatientEverythingService : IPatientEverythingService
             return new SearchResult(searchResultEntries, searchResult.ContinuationToken, searchResult.SortOrder, searchResult.UnsupportedSearchParameters);
         }
 
+        private void StoreSeeAlsoLinkInformationInToken(string resourceId, EverythingOperationContinuationToken token, IReadOnlyCollection<SearchResultEntry> searchResultEntries)
+        {
+            SearchResultEntry patientResource = searchResultEntries.FirstOrDefault(s => string.Equals(s.Resource.ResourceTypeName, ResourceType.Patient.ToString(), StringComparison.Ordinal));
+
+            if (patientResource.Resource != null)
+            {
+                ResourceElement element = _resourceDeserializer.Deserialize(patientResource.Resource);
+                Patient patient = element.ToPoco<Patient>();
+
+                List<Patient.LinkComponent> links = patient.Link;
+
+                if (links != null)
+                {
+                    foreach (Patient.LinkComponent link in links)
+                    {
+                        ReferenceSearchValue referenceSearchValue = _referenceSearchValueParser.Parse(link.Other.Reference);
+
+                        // If the link does not point back to the current patient
+                        if (!string.Equals(referenceSearchValue.ResourceId, resourceId, StringComparison.Ordinal))
+                        {
+                            if (link.Type == Patient.LinkType.ReplacedBy)
+                            {
+                                throw new EverythingOperationException(string.Format(Core.Resources.EverythingOperationResourceIrrelevant, resourceId, referenceSearchValue.ResourceId), HttpStatusCode.BadRequest);
+                            }
+
+                            if (link.Type == Patient.LinkType.Seealso)
+                            {
+                                // Links can be of type Patient or RelatedPerson - only attempt to run the $everything operation on Patient resources
+                                if (string.Equals(referenceSearchValue.ResourceType, ResourceType.Patient.ToString(), StringComparison.Ordinal))
+                                {
+                                    token.AddSeeAlsoLink(referenceSearchValue.ResourceId);
```

Comment:
>AddSeeAlsoLink [](http://example.com/codeflow?start=42&length=14)

This has potential to blow up (we put that into url, and url are limited resource, while there is no limitation on having lot of seealso links for patient).

Can we have LastSeeAlsoId and just pick patient greater than it?

Reply:
I was waiting for someone to comment on this... 😊. I agree. Used a hashset to avoid dups in 45267d2.

Reply:
> I was waiting for someone to comment on this... 😊. I agree. Used a hashset to avoid dups in [45267d2](https://github.com/microsoft/fhir-server/commit/45267d2aa3a2ae940fc786a1f73a8bf8ab1fbf18).

To check duplicates, can we use `token.SeeAlsoLinks.Contains(...)` directly? It looks we don't need to create an extra HashSet.

Reply:
Yes, good point. I think the linear check will make more sense once we limit the number of links. Working on this now!

Reply:
As written below, I ended up only storing the current link id in the token in a8f5b52.